### PR TITLE
tsdb: use unique uploader name while building tsdb index filename, same as boltdb-shipper

### PR DESCRIPTION
**Pull request by sandeepsukhani**
**What this PR does / why we need it**:
While building file names from `boltdb-shipper` and `tsdb`, we add the name of the uploader to it.
To avoid collisions in filenames when running multiple ingesters with the same name, we also add a nanosecond precise timestamp to the ingester name, which is persistent as the uploader name in the active index directory.

This is not being done for the tsdb index built from ingesters which can cause collisions in filenames when running multiple ingesters with the same name. I have moved the code to manage the uploader name to the index shipper config and used the same to share it with tsdb.

### CODE REVIEW

File: pkg/storage/stores/shipper/shipper_index_client.go [hunk 1]
```diff
@@ -90,7 +86,7 @@ func (i *indexClient) init(storageClient client.ObjectClient, limits downloads.L
 	}
 
 	if i.cfg.Mode != indexshipper.ModeReadOnly {
-		uploader, err := i.getUploaderName()
+		uploader, err := i.cfg.GetUniqueUploaderName()
```
Comment (kavirajk) [inline pkg/storage/stores/shipper/shipper_index_client.go:89]: Q: So this name is applied when only handling `tsdb` files. And boltdb index files are not impacted correct? just double checking.

File: pkg/storage/stores/indexshipper/shipper.go [hunk 2]
```diff
@@ -96,6 +99,33 @@ func (cfg *Config) Validate() error {
 	return storage.ValidateSharedStoreKeyPrefix(cfg.SharedStoreKeyPrefix)
 }
 
+func (cfg *Config) GetUniqueUploaderName() (string, error) {
```
Comment (kavirajk) [inline pkg/storage/stores/indexshipper/shipper.go:102]: nit: Can we small example how does the file name would look like as a comment description for this function?

**Review by kavirajk — APPROVED** [review]
LGTM. Thanks @sandeepsukhani . Jut one clarification and one minor nit :+1:

File: pkg/storage/stores/indexshipper/shipper.go (hunk 2 above)
Comment (sandeepsukhani) [reply pkg/storage/stores/indexshipper/shipper.go:102]: Filenames are handled by the underlying index builders i.e `boltdb-shipper` and `tsdb`. I will add some comments at appropriate places.

File: pkg/storage/stores/shipper/shipper_index_client.go (hunk 1 above)
Comment (sandeepsukhani) [reply pkg/storage/stores/shipper/shipper_index_client.go:89]: I have just copied the code to build the same uploader name from both `boltdb-shipper` and `tsdb`. There should not be any impact to `boltdb-shipper` file names.

**Pull request merged**